src/main.rs

```diff
@@ -555,7 +557,7 @@ fn main() -> Result<(), Box<dyn Error>> {
         .enable_all()
         .build()?;
     runtime.block_on(async {
-        let server = Server::bind(&opts.address).serve(service);
+        let server = Server::bind(&opts.address).http2_only(true).serve(service);
```

Comment:
Weird, the doc claims that http2 is supported by default:

> A listening HTTP server that accepts connections in both HTTP1 and HTTP2 by default.
> -- https://docs.rs/hyper/0.14.13/hyper/server/struct.Server.html

I think strictly requiring HTTP2 here might break quite a few clients

Reply:
Instead of directly setting this to `true`, can you add a new field to `Opts`, which defaults to `false` to keep the current default behavior?

Reply:
The problem here is `icx-proxy` is often invoked indirectly by `dfx`, so we don't get to control the command line options passed to it. Would you accept an environment variable configuration to turn this on?

Reply:
Can't we just change all the places where `dfx` runs `icx-proxy` (i.e. add the flag to those calls)?

Reply:
Adding a command-line option to icx-proxy, and adding parameters to `dfx start` and `dfx bootstrap` to control it, sounds like a better option.